packages/wellbeing-mock-data/wellbeing-mock-data.js

```diff
@@ -130,20 +130,106 @@ function createMockActivities() {
+  */
+function getRandomMoveInDate(start) {
+  var endDate = new Date();
+  var startDate = new Date();
+  startDate.setMonth(endDate.getMonth() - start);
+  moveInDate = randomDate(endDate, startDate);
+  return moveInDate;
+}
+
+  /*
+    This function assumes residents are already added.
+    then adds residency for every resident
+    Depending on the percentMovedOut some residents will have multiple residencies
+
+  */
+function createMockResidency(startingPoint, percentMovedOut) {
+  console.log("Creating Mock Residencies")
+  //get all residents
+  var residents = Residents.find().fetch();
+
+  /*  this is the number from where residents are moved out. So if the percentMovedout is 0.25
+      this index will be at the 75% percent point of the resident list
+      when the loop reaches this index it will move the residents randomly either to another home or just move them out completely
+  */
+  var indexWhereMovedOut = residents.length - Math.round(residents.length * percentMovedOut)
+  for (var i = 0; i < residents.length; i++) {
+    moveInDate = getRandomMoveInDate(startingPoint);
+    var args = { "residentId": residents[i]._id, 'homeId': residents[i].homeId, 'moveIn': moveInDate }
+    if (i >= indexWhereMovedOut) {
+      moveOutDate = randomDate(moveInDate, new Date());
+      args['moveOut'] = moveOutDate;
+      // some residents moved out permanently and some to another house >0.5 moved out permanently < 0.5 new house
+      var MovedOutPermanently = Math.random()
+      if (MovedOutPermanently < 0.5) {
+        var randomHome = getRandomHomeButExcludeCurrent(residents[i].homeId);
+        //move them into new house day after they left their old one( adding two days since the UTC 0:0:0 time)
+        var newMoveInDate = new Date(moveOutDate.getFullYear(), moveOutDate.getMonth(), moveOutDate.getDate() + 2);
+        var newArgs = { "residentId": residents[i]._id, 'homeId': randomHome, 'moveIn': newMoveInDate };
+        Residencies.insert(newArgs)
+      }
+    }
+  Residencies.insert(args);
+  }
+}
+
+
 Meteor.methods({
-  'createMockData': function() {
+  'createMockData': function (start, percentMovedOut) {
     createMockGroups();
     createMockRoles();
     createMockHomes();
     createMockResidents();
     createMockActivityTypes();
     createMockActivities();
+    createMockResidency(start, percentMovedOut);
+  },
+  'createMockGroups': function () {
+    createMockGroups();
+  },
+  'createMockRoles': function () {
+    createMockRoles();
+  },
+  'createMockHomes': function () {
+    createMockHomes();
+  },
+  'createMockResidents': function () {
+    createMockResidents();
+  },
+  'createActivityTypes': function () {
+    createMockHomes();
+  },
+  'createMockActivities': function () {
+    createMockActivities();
+  },
+  'createMockResidency': function (start, percentMovedOut) {
+createMockResidency(start, percentMovedOut);
   },
-  'removeAllData': function() {
+  'removeAllData': function () {
```

Comment:
I am just thinking that this `removeAllData` method seems dangerous. It should at least check that the current user is an admin.

Reply:
I didn't even realize this could be ran from the browser. My bad! I've been running this from meteor shell. I'll add check to it.